### PR TITLE
Custom help content for "Enter WordPress.com email" screen

### LINKED ISSUE #7572: [Analytics]: Same `Flow` and `Step` values for "Enter Store Address" and "Enter WPCOM email" screens
### Issue
While working on https://github.com/woocommerce/woocommerce-ios/issues/7547 I noticed that both "Enter Store Address" and "Enter WordPress.com email address" screens have the same `Flow` and `Step` values. 

This makes it impossible to differentiate between the following screens in analytics events. 

| Enter store address | Enter email address |
| --- | --- |
| Simulator Screen Shot - iPhone 11 Pro - 2022-08-26 at 05 59 09 | Simulator Screen Shot - iPhone 11 Pro - 2022-08-26 at 05 59 38 | 

### Steps to reproduce
1. Launch the app & log out if needed.
2. Skip the login prologue if needed.
3. Tap on "Enter Your Store Address" button
4. Observe that the following event is tracked upon landing in the screen to enter store address.
```
Tracked unified_login_step, properties: [AnyHashable("source"): "default", AnyHashable("step"): "start", AnyHashable("flow"): "login_site_address"]
```
5. Enter a valid store address and tap "Continue"
6. Observe that the following event is tracked upon landing in the screen to enter email address.
```
Tracked unified_login_step, properties: [AnyHashable("step"): "start", AnyHashable("source"): "default", AnyHashable("flow"): "login_site_address"]
```
7. Observe that both the tracked `unified_login_step` events have the same properties which makes them impossible to differentiate. 

### Expected behaviour
For the "Enter email address" screen Android uses ["enter_email_address"](https://github.com/woocommerce/woocommerce-android/blob/trunk/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/UnifiedLoginTracker.kt#L154) as `Step` value and we should use the same for iOS.

This needs to be fixed in [the WordPressAuthenticator-iOS library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/).

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #7572
Part of: #7547 

- [ ] ⚠️ Please review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/674 ⚠️
- [ ] ⚠️ @selanthiraiyan update Podfile after https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/674 is merged and released ⚠️

### Description
As a part of improving the help center content, this PR adds custom content with FAQs and Answers for the "Enter WordPress.com email" screen at the following URL.

https://woocommerce.com/document/android-ios-apps-login-help-faq/#enter-wordpress-com-email-address-login-using-store-address-flow

Internal ref - pe5sF9-sZ-p2

#### For #7572 
1. Install and launch the app.
1. Tap `Skip` to reach the login prologue screen.
1. Tap the `Enter Your Store Address` button.
1. In Xcode debug console, observe that the following event is tracked 
```
Tracked unified_login_step, properties: [AnyHashable("flow"): "login_site_address", AnyHashable("step"): "start", AnyHashable("source"): "default"]
```
1. Enter a valid store address and tap the `Continue` button
1. In Xcode debug console, observe that the following event is tracked upon landing in enter WPCOM email screen
```
Tracked unified_login_step, properties: [AnyHashable("step"): "enter_email_address", AnyHashable("flow"): "login_site_address", AnyHashable("source"): "default"]
```

#### For #7547 
1. Install and launch the app.
1. Tap `Skip` to reach the login prologue screen.
1. Tap the `Enter Your Store Address` button.
1. Enter a valid store address and tap the `Continue` button
1. Tap `Help` -> `Help Center`
1. Observe that you are directed to https://woocommerce.com/document/android-ios-apps-login-help-faq/#enter-wordpress-com-email-address-login-using-store-address-flow in a web view
1. In Xcode debug console observe that the following event is tracked 
```
Tracked support_help_center_viewed, properties: [AnyHashable("help_content_url"): "https://woocommerce.com/document/android-ios-apps-login-help-faq/#enter-wordpress-com-email-address-login-using-store-address-flow", AnyHashable("source_flow"): "login_site_address", AnyHashable("source_step"): "enter_email_address"]
```

### Screenshots

https://user-images.githubusercontent.com/524475/186804166-13010858-4d1c-4854-b959-f4b6837743c2.mov

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/System/WooConstants.swift [hunk 1]
```diff
@@ -81,6 +81,11 @@ extension WooConstants {
         ///
         case helpCenterForEnterStoreAddress = "https://woocommerce.com/document/android-ios-apps-login-help-faq/#enter-store-address"
 
+        /// Help Center for "Enter WordPress.com email" screen
+        ///
+        // swiftlint:disable:next line_length
```
Comment (jaclync) [inline WooCommerce/Classes/System/WooConstants.swift:86]: nice, TIL!

File: WooCommerce/Classes/System/WooConstants.swift [hunk 2]
```diff
@@ -81,6 +81,11 @@ extension WooConstants {
...
+        case helpCenterForEnterWPCOMEmail = "https://woocommerce.com/document/android-ios-apps-login-help-faq/#enter-wordpress-com-email-address-login-using-store-address-flow"
```
Comment (jaclync) [inline WooCommerce/Classes/System/WooConstants.swift]: nit: it looks like this URL is specific to the site address flow, and we are going to add a similar section about WP.com email from the WP.com flow. Maybe we want to specify this in the name? something like `helpCenterForWPCOMEmailWithSiteAddress`/`helpCenterForWPCOMEmailFromSiteAddress`

File: WooCommerce/Classes/Model/CustomHelpCenterContent.swift [hunk 3]
```diff
@@ -27,8 +27,10 @@ extension CustomHelpCenterContent {
     ///
     init?(step: AuthenticatorAnalyticsTracker.Step, flow: AuthenticatorAnalyticsTracker.Flow) {
         switch step {
-        case .start where flow == .loginWithSiteAddress:
+        case .start where flow == .loginWithSiteAddress: // Enter Store Address screen
             url = WooConstants.URLs.helpCenterForEnterStoreAddress.asURL()
+        case .enterEmailAddress where flow == .loginWithSiteAddress: // Enter WordPress.com email screen
```
Comment (jaclync) [inline WooCommerce/Classes/Model/CustomHelpCenterContent.swift]: super nit: might be worth mentioning the site address flow to differentiate it from the WP.com flow
```suggestion
        case .enterEmailAddress where flow == .loginWithSiteAddress: // Enter WordPress.com email screen from store address flow
```

File: RELEASE-NOTES.txt [hunk 4]
```diff
@@ -6,6 +6,7 @@
 - [*] Help center: Added custom help center web page with FAQs for "Enter Store Address" screen. [https://github.com/woocommerce/woocommerce-ios/pull/7553]
 - [*] In-Person Payments: The plugin selection is saved correctly after multiple onboarding prompts. [https://github.com/woocommerce/woocommerce-ios/pull/7544]
 - [*] In-Person Payments: A new prompt to enable `Pay in Person` for your store's checkout, to accept In-Person Payments for website orders [https://github.com/woocommerce/woocommerce-ios/issues/7474]
+- [*] Help center: Added custom help center web page with FAQs for "Enter WordPress.com email" screen. [https://github.com/woocommerce/woocommerce-ios/pull/7573]
```
Comment (jaclync) [inline RELEASE-NOTES.txt]: nit: might be nice batching this entry with the previous Help center entry to make it easier for our quality engineers to combine the testing?

**Review by jaclync — APPROVED** [review]
LGTM! :shipit: The nits are very minor and non-blocking

File: RELEASE-NOTES.txt (hunk 4 above)
Comment (selanthiraiyan) [reply RELEASE-NOTES.txt]: Thanks for the tip. Done in https://github.com/woocommerce/woocommerce-ios/pull/7573/commits/e0586c30f73cbb3031cb8976fcb309090df168aa

File: WooCommerce/Classes/Model/CustomHelpCenterContent.swift (hunk 3 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/Model/CustomHelpCenterContent.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/7573/commits/27fd7007e2dcaf1e3a5ace1467b31264ac326c28

File: WooCommerce/Classes/System/WooConstants.swift (hunk 2 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/System/WooConstants.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/7573/commits/7bf9cbc7af22a414494f808036ba7a58c8c0e7d3

**Pull request merged**